extensions/graphite-emitter/src/main/java/io/druid/emitter/graphite/GraphiteEmitter.java

```diff
@@ -0,0 +1,187 @@
+                                    "GraphiteEmitter-%s")
+                                .build()
+  ); // Thread pool of two in order to schedule flush runnable
+
+  public GraphiteEmitter(@NotNull GraphiteEmitterConfig graphiteEmitterConfig, @NotNull ObjectMapper mapper)
+  {
+    this.mapper = mapper;
+    this.graphiteEmitterConfig = graphiteEmitterConfig;
+    this.graphiteEventConverter = new GraphiteEventConverter(
+        graphiteEmitterConfig.getNamespacePrefix(),
+        graphiteEmitterConfig.getIgnoreHostname(),
+        graphiteEmitterConfig.getIgnoreServiceName(),
+        graphiteEmitterConfig.getWhiteListDimsMapper()
+    );
+    this.eventsQueue = new LinkedBlockingQueue(graphiteEmitterConfig.getMaxQueueSize());
+  }
+
+  @Override
+  public void start()
+  {
+    log.info("Starting Graphite Emitter.");
+    synchronized (started) {
+      if (!started.get()) {
+        exec.scheduleAtFixedRate(
+            new ConsumerRunnable(),
+            graphiteEmitterConfig.getFlushPeriod(),
+            graphiteEmitterConfig.getFlushPeriod(),
+            TimeUnit.MILLISECONDS
+        );
+        started.set(true);
+      }
+    }
+  }
+
+
+  @Override
+  public void emit(Event event)
+  {
+    if (!started.get()) {
+      throw new ISE("WTF emit was called while service is not started yet");
+    }
+
+    if (event instanceof ServiceMetricEvent) {
+      final ServiceMetricEvent serviceMetricEvent = (ServiceMetricEvent) event;
+      final GraphiteEvent graphiteEvent = graphiteEventConverter.druidMetricToGraphite(serviceMetricEvent);
+      if (graphiteEvent == null) {
+        return;
+      }
+      try {
+        final boolean isSuccessful = eventsQueue.offer(graphiteEvent, DEFAULT_PUT_GET_TIMEOUT, TimeUnit.MILLISECONDS);
+        if (!isSuccessful) {
+          log.error(
+              "Throwing event [%s] on the floor Graphite queue is full please increase the capacity or/and the consumer frequency",
+              mapper.writeValueAsString(event)
+          );
+        }
+      }
+      catch (InterruptedException | JsonProcessingException e) {
+        Thread.interrupted();
+        log.error(e, e.getMessage(),Thread.interrupted());
+      }
+    } else if (graphiteEmitterConfig.getLogAlerts() && event instanceof AlertEvent) {
+      try {
+        log.warn(mapper.writeValueAsString(event));
+      }
+      catch (JsonProcessingException e) {
+        log.error(e, "error while serializing alert event[%s]", ((AlertEvent) event).getDescription());
+      }
+    }
+  }
+
+  private class ConsumerRunnable implements Runnable
+  {
+    private PickledGraphite pickledGraphite = new PickledGraphite(
+        graphiteEmitterConfig.getHostname(),
+        graphiteEmitterConfig.getPort(),
+        graphiteEmitterConfig.getBatchSize()
+    );
+
+    @Override
+    public void run()
+    {
+      try {
+
+        if (!pickledGraphite.isConnected()) {
+          log.debug("trying to connect to graphite server");
```

Comment:
this can be an "info" log,just to inform the user in case graphite server is non reachable